### ISSUE TITLE
make[3]: mips-elf-gcc: Command not found

### ISSUE THREAD

**Issue by overgara2012** [Ai-Thinker-Open/GPRS_C_SDK#296]
## 1. SDK version（SDK 版本）

{V2112

}

---

## 2. In what kind of operation problems appear, and how to reproduce the problem ?（什么样的操作步骤问题会出现，是否是稳定复现，如何复现问题？）

{
omarv@omarv-msi:~/Downloads/A9G/projects/GPRS_C_SDK$ ./build.sh demo gpio release
param number:2
compileMode:release
compile project gpio
compile path /home/omarv/Downloads/A9G/projects/GPRS_C_SDK/demo/gpio
core number:8
Build host is LINUX

MAKE              init
CC                sdk_init.c
make[3]: mips-elf-gcc: Command not found
/home/omarv/Downloads/A9G/projects/GPRS_C_SDK/platform/compilation/cust_rules.mk:958: recipe for target '/home/omarv/Downloads/A9G/projects/GPRS_C_SDK/build/gpio/init/obj/release/sdk_init.o' failed
make[3]: *** [/home/omarv/Downloads/A9G/projects/GPRS_C_SDK/build/gpio/init/obj/release/sdk_init.o] Error 127
make[3]: *** Waiting for unfinished jobs....
/home/omarv/Downloads/A9G/projects/GPRS_C_SDK/platform/compilation/cust_rules.mk:975: recipe for target 'dependencies' failed
make[2]: *** [dependencies] Error 2
/home/omarv/Downloads/A9G/projects/GPRS_C_SDK/platform/compilation/cust_rules.mk:718: recipe for target '/home/omarv/Downloads/A9G/projects/GPRS_C_SDK/build/gpio//gpio.srec' failed
make[1]: *** [/home/omarv/Downloads/A9G/projects/GPRS_C_SDK/build/gpio//gpio.srec] Error 2
/home/omarv/Downloads/A9G/projects/GPRS_C_SDK/platform/compilation/cust_rules.mk:371: recipe for target 'lod' failed
make: *** [lod] Error 2
=== Build Time: 0s at 2019-01-17 03:22:54 ===
!!!!!!!!!!!!!!!!!!!!
   BUILD FAILED
!!!!!!!!!!!!!!!!!!!!
omarv@omarv-msi:~/Downloads/A9G/projects/GPRS_C_SDK$ 

}

---

**Comment by overgara2012** [comment]
omarv@omarv-msi:~/Downloads/A9G/projects/GPRS_C_SDK$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.1 LTS
Release:	18.04
Codename:	bionic
omarv@omarv-msi:~/Downloads/A9G/projects/GPRS_C_SDK$

**Comment by pulkin** [comment]
1. Make sure you ran **setup.sh** from CSDTK
2. Make sure you have `GPRS CSDTK` in your bashrc
3. Make sure you opened a new terminal session after this

**Comment by overgara2012** [comment]
I follow all recommnedation steps (run setup.sh, path to GPRS and CSDTK in .bashrc, source .bashrc, open new terminal,  but again same error:

If I edit the .bashrc with nano, this this that show:


export PATH="$PATH":.//home/omarv/software/CSDTK/
export PATH="$PATH":.//home/omarv/projects/GPRS_C_SDK

###############GPRS CSDTK#############
alias work='source .//cygenv.sh'
export GPRS_PROJ_ROOT=/home/omarv/projects
export PATH=$PATH:.//bin:.//mingw32/usr/bin:.//mips-rda-elf/bin:.//rv32-elf/bin
export PATH=$PATH:.//cooltools
export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:.//lib:.//mingw32/usr/lib
############GPRS CSDTK END############

the //cooltools directory is not in the github zip file

-------------------------------------------------------------------------------

omarv@omarv-msi:~/projects/GPRS_C_SDK$ ./build.sh demo gpio
param number:2
compileMode:debug
compile project gpio
compile path /home/omarv/projects/GPRS_C_SDK/demo/gpio
core number:8
Build host is LINUX

MAKE              init
CC                sdk_init.c
make[3]: mips-elf-gcc: Command not found
/home/omarv/projects/GPRS_C_SDK/platform/compilation/cust_rules.mk:958: recipe for target '/home/omarv/projects/GPRS_C_SDK/build/gpio/init/obj/debug/sdk_init.o' failed
make[3]: *** [/home/omarv/projects/GPRS_C_SDK/build/gpio/init/obj/debug/sdk_init.o] Error 127
make[3]: *** Waiting for unfinished jobs....
/home/omarv/projects/GPRS_C_SDK/platform/compilation/cust_rules.mk:975: recipe for target 'dependencies' failed
make[2]: *** [dependencies] Error 2
/home/omarv/projects/GPRS_C_SDK/platform/compilation/cust_rules.mk:718: recipe for target '/home/omarv/projects/GPRS_C_SDK/build/gpio//gpio.srec' failed
make[1]: *** [/home/omarv/projects/GPRS_C_SDK/build/gpio//gpio.srec] Error 2
/home/omarv/projects/GPRS_C_SDK/platform/compilation/cust_rules.mk:371: recipe for target 'lod' failed
make: *** [lod] Error 2
=== Build Time: 0s at 2019-02-01 09:09:48 ===
!!!!!!!!!!!!!!!!!!!!
   BUILD FAILED
!!!!!!!!!!!!!!!!!!!!
omarv@omarv-msi:~/projects/GPRS_C_SDK$ 

someone has any advice ?,  in windows10 it works like a charm.., but I hate have to use windows...

**Comment by pulkin** [comment]
This is wrong. I do not know why it failed to determine the full path. You should fix it manually or try running `setup.sh` once more. This is my setup for your reference:
```bash
./setup.sh ./ ~/Projects/a9g/projects/
```
```bash
###############GPRS CSDTK#############
alias work='source /home/pulkin/Projects/a9g/CSDTK/cygenv.sh'
export GPRS_PROJ_ROOT=/home/pulkin/Projects/a9g/projects
export PATH=$PATH:/home/pulkin/Projects/a9g/CSDTK/bin:/home/pulkin/Projects/a9g/CSDTK/mingw32/usr/bin:/home/pulkin/Projects/a9g/CSDTK/mips-rda-elf/bin:/home/pulkin/Projects/a9g/CSDTK/rv32-elf/bin
export PATH=$PATH:/home/pulkin/Projects/a9g/CSDTK/cooltools
export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/home/pulkin/Projects/a9g/CSDTK/lib:/home/pulkin/Projects/a9g/CSDTK/mingw32/usr/lib
############GPRS CSDTK END############
```

**Comment by overgara2012** [comment]
I already fix it. 

thanks anyway...

But now I have another issue: library missing:

omarv@omarv-msi:~/projects/GPRS_C_SDK$ ./build.sh demo gpio
param number:2
compileMode:debug
compile project gpio
compile path /home/omarv/projects/GPRS_C_SDK/demo/gpio
core number:8
Build host is LINUX

MAKE              init
CC                sdk_init.c
/home/omarv/software/CSDTK/bin/../libexec/gcc/mips-elf/4.4.2/cc1: error while loading shared libraries: libcloog.so.0: cannot open shared object file: No such file or directory
as: unrecognized option '-G0'
/home/omarv/projects/GPRS_C_SDK/platform/compilation/cust_rules.mk:958: recipe for target '/home/omarv/projects/GPRS_C_SDK/build/gpio/init/obj/debug/sdk_init.o' failed
make[3]: *** [/home/omarv/projects/GPRS_C_SDK/build/gpio/init/obj/debug/sdk_init.o] Error 1
make[3]: *** Waiting for unfinished jobs....
/home/omarv/projects/GPRS_C_SDK/platform/compilation/cust_rules.mk:975: recipe for target 'dependencies' failed
make[2]: *** [dependencies] Error 2
/home/omarv/projects/GPRS_C_SDK/platform/compilation/cust_rules.mk:718: recipe for target '/home/omarv/projects/GPRS_C_SDK/build/gpio//gpio.srec' failed
make[1]: *** [/home/omarv/projects/GPRS_C_SDK/build/gpio//gpio.srec] Error 2
/home/omarv/projects/GPRS_C_SDK/platform/compilation/cust_rules.mk:371: recipe for target 'lod' failed
make: *** [lod] Error 2
=== Build Time: 0s at 2019-02-01 10:10:48 ===
!!!!!!!!!!!!!!!!!!!!
   BUILD FAILED
!!!!!!!!!!!!!!!!!!!!
omarv@omarv-msi:~/projects/GPRS_C_SDK$

**Comment by pulkin** [comment]
Obviously you did not fix it completely.
```bash
~ locate libcloog.so
/home/pulkin/Projects/a9g/CSDTK/lib/libcloog.so.0
/home/pulkin/Projects/a9g/CSDTK/lib/libcloog.so.0.0.0
```
You should have this folder in LD_LIBRARY_PATH.

**Comment by niwantha33** [comment]
and also check the user permission
